families/supplychain/restapi/openapi.yaml

```diff
@@ -0,0 +1,377 @@
+          description: A partial address to filter leaves by
+        - $ref: "#/parameters/count"
+        - $ref: "#/parameters/min"
+        - $ref: "#/parameters/max"
+        - $ref: "#/parameters/sort"
+      responses:
+        200:
+          description: Successfully retrieved state data
+          schema:
+            properties:
+              data:
+                type: array
+                items:
+                  $ref: "#/definitions/Agent"
+              head:
+                $ref: "#/definitions/Head"
+              link:
+                $ref: "#/definitions/Link"
+              paging:
+                $ref: "#/definitions/Paging"
+        400:
+          $ref: "#/responses/400BadRequest"
+        500:
+          $ref: "#/responses/500ServerError"
+        503:
+          $ref: "#/responses/503ServiceUnavailable"
+  /agents/{agent_id}:
+    get:
+      summary: Fetches a particular agent from the current state
+      parameters:
+        - $ref: "#/parameters/head"
+      responses:
+        200:
+          description: Successfully fetched leaves
+          schema:
+            properties:
+              data:
+                $ref: "#/definitions/Agent"
+              head:
+                $ref: "#/definitions/Head"
+              link:
+                $ref: "#/definitions/Link"
+        400:
+          $ref: "#/responses/400BadRequest"
+        404:
+          $ref: "#/responses/404NotFound"
+        500:
+          $ref: "#/responses/500ServerError"
+        503:
+          $ref: "#/responses/503ServiceUnavailable"
+  /applications/:
+    get:
+      summary: Fetches the data for the current set of applications in state
+      description: >
+        Fetches a paginated list of leaves for the current state, or relative
+        to a particular head block. Using the `address` filter parameter, will
+        narrow the list to any leaves that have an address beginning with the
+        characters specified.
+      parameters:
+        - $ref: "#/parameters/head"
+        - name: address
+          in: query
+          type: string
+          description: A partial address to filter leaves by
+        - $ref: "#/parameters/count"
+        - $ref: "#/parameters/min"
+        - $ref: "#/parameters/max"
+        - $ref: "#/parameters/sort"
+      responses:
+        200:
+          description: Successfully retrieved state data
+          schema:
+            properties:
+              data:
+                type: array
+                items:
+                  $ref: "#/definitions/Application"
+              head:
+                $ref: "#/definitions/Head"
+              link:
+                $ref: "#/definitions/Link"
+              paging:
+                $ref: "#/definitions/Paging"
+        400:
+          $ref: "#/responses/400BadRequest"
+        500:
+          $ref: "#/responses/500ServerError"
+        503:
+          $ref: "#/responses/503ServiceUnavailable"
+  /applications/{record_id}:
```

Comment:
Applications are stored by record id. I have added text to the descriptions to clarify this.